sparkly/catalog.py

```diff
@@ -26,6 +27,62 @@ def __init__(self, spark):
         """
         self._spark = spark
 
+    def create_table(self, table_name, path=None, source=None, schema=None, **options):
```

Comment:
Instead of implementing similar functionality in internal plugins, why not add the metadata copying here after this is merged?

Also I'm not sure why we should explain the difference with `saveAsTable`. Imo pyspark api documentation is enough since we mention explicitly that this extends `createTable/createExternalTable`.